packages/data-stores/src/domain-suggestions/selectors.ts

```diff
@@ -11,78 +11,90 @@ import { DomainSuggestionQuery } from './types';
+		search: string,
+		options: DomainSuggestionSelectorOptions = {}
+	) {
+		const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
 
-export const isLoadingDomainSuggestions = (
-	_state: State,
-	search: string,
-	options: DomainSuggestionSelectorOptions = {}
-) => {
-	const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
+		// We need to go through the `select` store to get the resolver action
+		return select( STORE_KEY ).__internalGetDomainSuggestions( normalizedQuery );
+	}
 
-	return select( 'core/data' ).isResolving( STORE_KEY, '__internalGetDomainSuggestions', [
-		normalizedQuery,
-	] );
-};
+	function isLoadingDomainSuggestions(
+		_state: State,
+		search: string,
+		options: DomainSuggestionSelectorOptions = {}
+	) {
+		const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
 
-/**
- * Do not use this selector. It is for internal use.
- *
- * @private
- *
- * @param state Store state
- * @param queryObject Normalized object representing the query
- * @returns suggestions
- */
-export const __internalGetDomainSuggestions = (
-	state: State,
-	queryObject: DomainSuggestionQuery
-) => {
-	return state.domainSuggestions[ stringifyDomainQueryObject( queryObject ) ];
-};
+		return select( 'core/data' ).isResolving( STORE_KEY, '__internalGetDomainSuggestions', [
+			normalizedQuery,
+		] );
+	}
 
-/**
- * Normalize domain query
- *
- * It's important to have a consistent, reproduceable representation of a domains query so that the result can be
- * stored and retrieved.
- *
- * @see client/state/domains/suggestions/utils.js
- * @see client/components/data/query-domains-suggestions/index.jsx
- *
- * @param search       Domain search string
- * @param queryOptions Optional paramaters for the query
- * @returns Normalized query object
- */
-function normalizeDomainSuggestionQuery(
-	search: string,
-	queryOptions: DomainSuggestionSelectorOptions
-): DomainSuggestionQuery {
-	return {
-		// Defaults
-		include_wordpressdotcom: queryOptions.only_wordpressdotcom || false,
-		include_dotblogsubdomain: false,
-		only_wordpressdotcom: false,
-		quantity: 5,
-		vendor: 'variation2_front',
+	/**
+	 * Normalize domain query
+	 *
+	 * It's important to have a consistent, reproduceable representation of a domains query so that the result can be
+	 * stored and retrieved.
+	 *
+	 * @see client/state/domains/suggestions/utils.js
+	 * @see client/components/data/query-domains-suggestions/index.jsx
+	 *
+	 * @param search       Domain search string
+	 * @param queryOptions Optional paramaters for the query
+	 * @returns Normalized query object
+	 */
+	function normalizeDomainSuggestionQuery(
+		search: string,
+		queryOptions: DomainSuggestionSelectorOptions
+	): DomainSuggestionQuery {
+		return {
+			// Defaults
+			include_wordpressdotcom: queryOptions.only_wordpressdotcom || false,
+			include_dotblogsubdomain: false,
+			only_wordpressdotcom: false,
+			quantity: 5,
+			vendor: vendor,
```

Comment:
This is the most relevant change here. The rest is wrapping all the selectors in a create function to capture the vendor in the closure.

I attempted to implement this as a class, which seemed like a nice way to encapsulate state and behavior 😏 but it didn't seem to play well with wp-data and I desisted.